src/Microsoft.ML.ImageAnalytics/ImageGrayscale.cs

```diff
@@ -218,6 +218,9 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                             g.DrawImage(src, srcRectangle, 0, 0, src.Width, src.Height, GraphicsUnit.Pixel, attributes);
                         }
                         Contracts.Assert(dst.Width == src.Width && dst.Height == src.Height);
+
+                        src.Dispose();
+                        src = null;
```

Comment:
So, we see the correct pattern above at line 205, where `dst` is disposed if non-`null`. So the getter should in principle be disposing. Is it not? If it is not it should be.

So this seems to be disposing potentially in *both* places, but this makes me think we just have a bad getter delegate somewhere.

Reply:
That is dst, this is src.

Reply:
The point is that 205 is showing the correct pattern, and we also have the dispose method. If this pattern is being adhered to everywhere, the dispose method is being composed correct, and the cursor is being disposed, `src` should be being disposed already. Investigate why this is not so.

---
In reply to: [275461358](https://github.com/dotnet/machinelearning/pull/3328#discussion_r275461358) [](ancestors = 275461358)